homeassistant/components/solarlog/entity.py

```diff
@@ -0,0 +1,71 @@
+"""Entities for SolarLog integration."""
+
+from __future__ import annotations
+
+from homeassistant.components.sensor import SensorEntityDescription
+from homeassistant.helpers.device_registry import DeviceInfo
+from homeassistant.helpers.update_coordinator import CoordinatorEntity
+from homeassistant.util import slugify
+
+from .const import DOMAIN
+from .coordinator import SolarLogCoordinator
+
+
+class SolarLogCoordinatorEntity(CoordinatorEntity[SolarLogCoordinator]):
+    """Base SolarLog Coordinator entity."""
+
+    _attr_has_entity_name = True
+
+    def __init__(
+        self,
+        coordinator: SolarLogCoordinator,
+        description: SensorEntityDescription,
+    ) -> None:
+        """Initialize the SolarLogCoordinator sensor."""
+        super().__init__(coordinator)
+
+        self._attr_unique_id = f"{coordinator.unique_id}-{description.key}"
+        self._attr_device_info = DeviceInfo(
+            manufacturer="Solar-Log",
+            model="Controller",
+            identifiers={(DOMAIN, coordinator.unique_id)},
+            name=coordinator.name,
+            configuration_url=coordinator.host,
+        )
+
+        self.entity_description = description
+
+
+class SolarLogInverterEntity(CoordinatorEntity[SolarLogCoordinator]):
+    """Base SolarLog inverter entity."""
+
+    _attr_has_entity_name = True
+
+    def __init__(
+        self,
+        coordinator: SolarLogCoordinator,
+        description: SensorEntityDescription,
+        device_id: int,
+    ) -> None:
+        """Initialize the SolarLogInverter sensor."""
+        super().__init__(coordinator)
+        self._attr_unique_id = f"{coordinator.unique_id}-{slugify(coordinator.solarlog.device_name(device_id))}-{description.key}"
+        self._attr_device_info = DeviceInfo(
+            manufacturer="Solar-Log",
+            model="Inverter",
+            identifiers={
+                (
+                    DOMAIN,
+                    f"{coordinator.unique_id}-{slugify(coordinator.solarlog.device_name(device_id))}",
+                )
+            },
+            name=coordinator.solarlog.device_name(device_id),
```

Comment:
What does device name return?

Reply:
A user-defined name of the device (can be set in SolarLog configuration)

Reply:
Then we should not have that in the unique id

Reply:
I double-checked in the manual (https://www.solar-log.com/fileadmin/user_upload/documents/Handbuecher_und_Inlet/de_DE/SolarLog_Base_Handbuch_6_x_DE.pdf) and on my SolarLog device and have to correct my above statement.
The device name is automatically detected and according to my understanding cannot be changed by the user. It's possible that it could be changed during the initial detection, but I also have not found a reference for this possibility in the manual either. The device name therefore is immutable.